src/reader/mod.rs

```diff
@@ -11,6 +11,109 @@ use crate::events::{BytesCData, BytesDecl, BytesEnd, BytesStart, BytesText, Even
 
 use memchr;
 
+macro_rules! builder_methods {
```

Comment:
I don't see why this needs to be a macro, and the commit message doesn't provide any wisdom.  Probably should be separate commits...